codegen/method.go

```diff
@@ -538,16 +551,115 @@ func (ms *MethodSpec) setTypeConverters(
 	downstreamRespFields := downstreamRespType.(*compile.StructSpec).Fields
 
 	respConverter := &TypeConverter{
-		Lines:  []string{},
-		Helper: h,
+		LineBuilder: LineBuilder{},
+		Helper:      h,
 	}
 
 	err = respConverter.GenStructConverter(downstreamRespFields, respFields)
 	if err != nil {
 		return err
 	}
 
-	ms.ConvertResponseGoStatements = respConverter.Lines
+	ms.ConvertResponseGoStatements = respConverter.GetLines()
+
+	return nil
+}
+
+func (ms *MethodSpec) setQueryParamStatements(
+	funcSpec *compile.FunctionSpec,
+) error {
+	statements := LineBuilder{}
+	structType := compile.FieldGroup(funcSpec.ArgsSpec)
+
+	for _, field := range structType {
+		realType := compile.RootTypeSpec(field.Type)
+		fieldName := field.Name
+		identifierName := camelCase(fieldName) + "Query"
+
+		httpRefAnnotation := field.Annotations[antHTTPRef]
+		// If a thrift field has a http.ref annotation then we
+		// should not read this field from query parameters.
+		if httpRefAnnotation != "" {
+			continue
+		}
+
+		okIdentifierName := camelCase(fieldName) + "Ok"
+		if field.Required {
+			statements.appendf("%s := req.CheckQueryValue(%q)",
+				okIdentifierName, fieldName,
+			)
+			statements.appendf("if !%s {", okIdentifierName)
+			statements.append("\treturn")
```

Comment:
For better error handling, `setQueryParamStatements` should construct a Go function with error return type, instead of lines of code. So that the endpoint code can properly log the error and control the flow of handling requests.